recipes/instinct-cpp/all/conanfile.py

```diff
@@ -0,0 +1,158 @@
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import apply_conandata_patches, copy, export_conandata_patches, get, replace_in_file, rm, rmdir
+from conan.tools.microsoft import check_min_vs, is_msvc, is_msvc_static_runtime
+from conan.tools.scm import Version
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class InstinctCppConan(ConanFile):
+    name = "instinct-cpp"
+    description = ("instinct.cpp provides ready to use alternatives to OpenAI Assistant API"
+                   "and built-in utilities for developing AI Agent applications (RAG, Chatbot, Code interpreter) powered by language models. "
+                   "Call it langchain.cpp if you like.")
+    license = "Apache-2.0"
+    url = "https://github.com/RobinQu/instinct.cpp"
+    homepage = "https://github.com/RobinQu/instinct.cpp"
+    topics = ("llm", "agent", "genai")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "with_duckdb": [True, False],
+        "with_exprtk": [True, False],
+        "with_pdfium": [True, False],
+        "with_duckx": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "with_duckdb": True,
+        "with_exprtk": True,
+        "with_pdfium": True,
+        "with_duckx": True,
+        "duckdb/*:with_httpfs": True,
+        "icu/*:with_extras": True,
+        "icu/*:data_packaging": "static"
+    }
+
+    @property
+    def _min_cppstd(self):
+        return 20
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "apple-clang": "15",
+            "clang": "17",
+            "gcc": "12"
+        }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+    def layout(self):
+        # src_folder must use the same source folder name the project
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("hash-library/8.0", transitive_headers=True)
+        self.requires("bshoshany-thread-pool/4.1.0", transitive_headers=True)
+
+        if self.options.with_duckdb:
+            self.requires("duckdb/0.10.2", transitive_headers=True)
+
+        self.requires("uriparser/0.9.7", transitive_headers=True)
+        self.requires("crossguid/0.2.2", transitive_headers=True)
+        self.requires("protobuf/5.27.0", transitive_headers=True)
+        self.requires("reactiveplusplus/2.1.1", transitive_headers=True)
+        self.requires("icu/74.1")
+        self.requires("tsl-ordered-map/1.1.0", transitive_headers=True)
+        self.requires("fmt/10.2.1", transitive_headers=True)
```

Comment:
The `transitive_headers` would only be needed if those includes were part of the public API of the project, usually there are a few, but it's a bit suprising that every dependency has it. Does every dependency leak to the consumers?